draft-ietf-quic-transport.md

```diff
@@ -2736,24 +2736,53 @@ streams to immediately become closed; open streams can be assumed to be
+These states SHOULD persist for at least three times the current Probe Timeout
+(PTO) interval as defined in {{QUIC-RECOVERY}}.
+
+An immediate close can be used after an application protocol has arranged to
+close a connection.  This might be after the application protocols negotiates a
+graceful shutdown.  The application protocol exchanges whatever messages that
+are needed to cause both endpoints to agree to close the connection, after which
+the application requests that the connection be closed.  When the application
+closes the connection, a CONNECTION_CLOSE frame with an appropriate error code
+will be used to signal closure.
+
+
+### Closing Connection State {#closing}
+
+An endpoint enters a closing period after initiating an immediate close;
+{{immediate-close}}.
 
 During the closing period, an endpoint that sends a CONNECTION_CLOSE frame
 SHOULD respond to any incoming packet that can be decrypted with another packet
-containing a CONNECTION_CLOSE frame.  Such an endpoint SHOULD limit the number
-of packets it generates containing a CONNECTION_CLOSE frame.  For instance, an
-endpoint could wait for a progressively increasing number of received packets or
-amount of time before responding to a received packet.
-
-An endpoint is allowed to drop the packet protection keys when entering the
-closing period ({{draining}}) and send a packet containing a CONNECTION_CLOSE in
-response to any UDP datagram that is received.  However, an endpoint without the
-packet protection keys cannot identify and discard invalid packets.  To avoid
-creating an unwitting amplification attack, such endpoints MUST limit the
-cumulative size of packets containing a CONNECTION_CLOSE frame to 3 times the
-cumulative size of the packets that cause those packets to be sent.  To minimize
-the state that an endpoint maintains for a closing connection, endpoints MAY
-send the exact same packet.
+containing a CONNECTION_CLOSE frame. An endpoint that is closing is not
+required to process the frames contained in packets. Such an endpoint SHOULD
+limit the number of packets it generates containing a CONNECTION_CLOSE frame.
+For instance, an endpoint could wait for a progressively increasing number of
+received packets or amount of time before responding to received packets.
+
+In the closing state, an endpoint retains only enough information to generate a
+packet containing a CONNECTION_CLOSE frame and to identify packets as belonging
+to the connection. The endpoint's selected connection ID and the QUIC version
+are sufficient information to identify packets for a closing connection; an
+endpoint can discard all other connection state. An endpoint MAY retain packet
+protection keys for incoming packets to allow it to read and process a
+CONNECTION_CLOSE frame.
+
+An endpoint MAY drop packet protection keys when starting the closing period
+and send a packet containing a CONNECTION_CLOSE in response to any UDP datagram
```

Comment:
This changes to use MAY instead of "is allowed to".

Otherwise, this changes only to remove the self-reference (and s/entering/starting).